### PR TITLE
Update project creation success screen and next steps modal

### PR DESCRIPTION
**Pull request by filipvvv**
## What does this PR do and why?

Update project creation success screen and next steps modal. Clarify language and add CTA to join the Discord.

Would love some help testing this one, /create dev mode proved too intensive for my machine. Will test in preview.

**Update:** tested this and it's working well.

https://user-images.githubusercontent.com/96905094/213941472-3ac241de-5fc4-42cb-b6fb-511160e1245c.mp4

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [x] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by tomquirk** [comment]
@filipvvv i'm lovin it. WDYT about stacking the buttons vertically instead?

Screenshot 2023-01-23 at 10 58 50 AM

File: src/components/Create/components/pages/ReviewDeploy/components/DeploySuccess/DeploySuccess.tsx [hunk 1]
```diff
@@ -60,12 +60,20 @@ export const DeploySuccess = ({ projectId }: { projectId: number }) => {
       <div className="pt-8 text-4xl font-medium">Congratulations!</div>
       <div className="mt-6 text-xl font-normal">{deployGreeting}</div>
       <div className="flex gap-2 pt-16">
+        <ExternalLink href="https://discord.gg/juicebox">
+          <Button>
+            <Space>
+              <RocketOutlined />
+              <Trans> Share on Discord</Trans>
+            </Space>
+          </Button>
```
Comment (tomquirk) [inline src/components/Create/components/pages/ReviewDeploy/components/DeploySuccess/DeploySuccess.tsx:69]: **suggestion**: Not related to your PR but while we're here, would you mind updating all these buttons to:
- use the `icon` prop
- wrap the `Trans` tag in `span` (for formatting purposes)
- remove any leading space `' '` character from within the `Trans`'s
🙏 🙏 

```suggestion
          <Button icon={<RocketOutlined/>}>
              <span>
                <Trans>Share on Discord</Trans>
              </span>
          </Button>
```

**Comment by filipvvv** [comment]
@aeolianeth implemented your suggestions, switched to vertical buttons, and cleaned up styling/imports.

**Review by tomquirk — APPROVED** [review]
LGTM @filipvvv thanks for making the changes!

**Pull request merged**